pkg/helpers/apply/apply_suite_test.go

```diff
@@ -0,0 +1,249 @@
+	Expect(err).NotTo(HaveOccurred())
+
+	clusterv1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	restConfig = cfg
+	close(done)
+}, 60)
+
+var _ = AfterSuite(func() {
+	By("tearing down the test environment")
+
+	err := testEnv.Stop()
+	Expect(err).ToNot(HaveOccurred())
+})
+
+var _ = Describe("setOwnerRef", func() {
+	It("Add OwnerRef to core item", func() {
+		var nsOwner *corev1.Namespace
+		By("Creating ns owner", func() {
+			nsOwner = &corev1.Namespace{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "Namespace",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "my-ns-owner-1",
+				},
+			}
+			var err error
+			nsOwner, err = kubeClient.CoreV1().Namespaces().Create(context.TODO(), nsOwner, metav1.CreateOptions{})
+			Expect(err).To(BeNil())
+		})
+		By("setReferenceOwner", func() {
+			reader := scenario.GetScenarioResourcesReader()
+			applierBuilder := &ApplierBuilder{}
+			applier := applierBuilder.
+				WithClient(kubeClient, apiExtensionsClient, dynamicClient).
+				WithTemplateFuncMap(FuncMap()).WithOwner(nsOwner, false, false, scheme.Scheme).
+				Build()
+
+			_, err := applier.ApplyDirectly(reader, nil, false, "", "ownerref/ns.yaml")
+			Expect(err).To(BeNil())
+		})
+		By("Checking Ownerref", func() {
+			ns, err := kubeClient.CoreV1().Namespaces().Get(context.TODO(), "my-ns", metav1.GetOptions{})
+			Expect(err).To(BeNil())
+			Expect(len(ns.GetOwnerReferences())).To(Equal(1))
+			Expect(ns.OwnerReferences[0].APIVersion).To(Equal(nsOwner.APIVersion))
+			Expect(ns.OwnerReferences[0].Kind).To(Equal(nsOwner.Kind))
+			Expect(ns.OwnerReferences[0].Name).To(Equal(nsOwner.Name))
+			Expect(ns.OwnerReferences[0].UID).To(Equal(nsOwner.UID))
+			Expect(ns.OwnerReferences[0].Controller).To(BeNil())
+			Expect(ns.OwnerReferences[0].BlockOwnerDeletion).To(BeNil())
+		})
+	})
+	It("Add OwnerRef to CRD item", func() {
+		var managedClusterOwner *clusterv1.ManagedCluster
+		By("Creating cluster owner", func() {
+			managedClusterOwner = &clusterv1.ManagedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "my-managedcluster-owner-1",
+				},
+			}
+			var err error
+			managedClusterOwner, err = clusterClient.ClusterV1().ManagedClusters().Create(context.TODO(), managedClusterOwner, metav1.CreateOptions{})
+			Expect(err).To(BeNil())
+		})
+		By("setReferenceOwner", func() {
+			reader := scenario.GetScenarioResourcesReader()
+			applierBuilder := &ApplierBuilder{}
+			applier := applierBuilder.
+				WithClient(kubeClient, apiExtensionsClient, dynamicClient).
+				WithTemplateFuncMap(FuncMap()).WithOwner(managedClusterOwner, false, false, scheme.Scheme).
+				Build()
+
+			_, err := applier.ApplyCustomResources(reader, nil, false, "", "ownerref/managedcluster.yaml")
+			Expect(err).To(BeNil())
+		})
+		By("Checking Ownerref", func() {
+			mc, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(), "my-managedcluster", metav1.GetOptions{})
```

Comment:
This seems to add a onwer of one mc to another?

Reply:
yes it is just and example, I used 2 mc, could be one mc and something else. it doesn't matter which CRD I use... it is just to test that one resource becomes the owner of another.